homeassistant/components/image_processing/tensorflow.py

```diff
@@ -0,0 +1,336 @@
+"""
+Component that performs TensorFlow classification on images.
+
+For a quick start, pick a pre-trained COCO model from:
+https://github.com/tensorflow/models/blob/master/research/object_detection/g3doc/detection_model_zoo.md
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/image_processing.tensorflow/
+"""
+import logging
+
+import voluptuous as vol
+
+from homeassistant.components.image_processing import (
+    CONF_CONFIDENCE, CONF_ENTITY_ID, CONF_NAME, CONF_SOURCE, PLATFORM_SCHEMA,
+    ImageProcessingEntity)
+from homeassistant.core import split_entity_id
+from homeassistant.helpers import template
+import homeassistant.helpers.config_validation as cv
+
+REQUIREMENTS = ['numpy==1.15.3', 'pillow==5.2.0', 'protobuf==3.6.1']
+
+_LOGGER = logging.getLogger(__name__)
+
+ATTR_MATCHES = 'matches'
+ATTR_TOTAL_MATCHES = 'total_matches'
+
+CONF_FILE_OUT = 'file_out'
+CONF_MODEL = 'model'
+CONF_GRAPH = 'graph'
+CONF_LABELS = 'labels'
+CONF_MODEL_DIR = 'model_dir'
+CONF_CATEGORIES = 'categories'
+CONF_CATEGORY = 'category'
+CONF_AREA = 'area'
+CONF_TOP = 'top'
+CONF_LEFT = 'left'
+CONF_BOTTOM = 'bottom'
+CONF_RIGHT = 'right'
+
+DEFAULT_MODEL_DIR = "/usr/src/app/tensorflow"
+DEFAULT_LABELS = ("{0}/object_detection/data/mscoco_label_map.pbtxt"
+                  .format(DEFAULT_MODEL_DIR))
+
+AREA_SCHEMA = vol.Schema({
+    vol.Optional(CONF_TOP, default=0): cv.small_float,
+    vol.Optional(CONF_LEFT, default=0): cv.small_float,
+    vol.Optional(CONF_BOTTOM, default=1): cv.small_float,
+    vol.Optional(CONF_RIGHT, default=1): cv.small_float
+})
+
+CATEGORY_SCHEMA = vol.Schema({
+    vol.Required(CONF_CATEGORY): cv.string,
+    vol.Optional(CONF_AREA): AREA_SCHEMA
+})
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Optional(CONF_FILE_OUT, default=[]):
+        vol.All(cv.ensure_list, [cv.template]),
+    vol.Required(CONF_MODEL): vol.Schema({
+        vol.Required(CONF_GRAPH): cv.isfile,
+        vol.Optional(CONF_LABELS, default=DEFAULT_LABELS): cv.isfile,
+        vol.Optional(CONF_MODEL_DIR, default=DEFAULT_MODEL_DIR): cv.isdir,
+        vol.Optional(CONF_AREA): AREA_SCHEMA,
+        vol.Optional(CONF_CATEGORIES, default=[]):
+            vol.All(cv.ensure_list, [vol.Any(
+                cv.string,
+                CATEGORY_SCHEMA
+            )])
+    })
+})
+
+
+def setup_platform(hass, config, add_entities, discovery_info=None):
+    """Set up the TensorFlow image processing platform."""
+    import sys
+
+    model_config = config.get(CONF_MODEL)
+    model_dir = model_config.get(CONF_MODEL_DIR)
+
+    # append custom model path to sys.path
+    sys.path.append(model_dir)
```

Comment:
Ehhhrrrrr

We should be able to specify the paths to look to tensorflow?

Reply:
There are some requirements that can not be installed via pip.  The object detection portion of the library is not included in the standard TensorFlow wheel.  This is to include the protobuf models and util functions created as a part of the tensorflow install shell script.